tools/ctl/ctl.c

```diff
@@ -30,6 +30,7 @@ struct ctl_data {
 	int out_fd;
 
 	/* cached buffer for input/output */
+	/* use unsigned int to ensure alignment: Warning on pointer arithmetic. */
```

Comment:
just thinking for the arm folk, if this is running on arm 32-bit, isn't an unsigned int 16bits? Shouldnt this be uint32_t?

Reply:
@dbaluta @paulstelian97 @afq984 @kuanhsuncheng any comments here for ARM32 ?

Reply:
> if this is running on arm 32-bit, isn't an unsigned int 16bits?

I believe it depends on the toolchain, but it's 4 on clang 11 and gcc 11: https://godbolt.org/z/bo8r5as1b

The comment `use unsigned int to ensure alignment` sounds a bit confusing to me. It doesn't ensure a misaligned pointer can't be stored into the `buffer` such as `buffer = (unsigned int*)(uintptr_t)3`.
Does it mean access to the `buffer` must be have an address of a multiple of 4?

Reply:
I have no idea since all the variants of i.MX8 are 64-bit.

Reply:
Agreed, I think we should in generate be defaulting to inttypes

Reply:
I did some update there.